gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -89,6 +89,130 @@ def _load_cadd_raw(cadd_tsv) -> hl.Table:
+
+    :return: Hail Table with SpliceAI scores for GRCh38.
+    """
+    logger.info("Importing vcf of SpliceAI scores into MT...")
+    recode = {f"{i}": f"chr{i}" for i in (list(range(1, 23)) + ["X", "Y"])}
+    spliceai_snvs = hl.import_vcf(
+        snvs_path,
+        force_bgz=True,
+        min_partitions=1000,
+        reference_genome="GRCh38",
+        contig_recoding=recode,
+        skip_invalid_loci=True,
+    )
+
+    spliceai_indels = hl.import_vcf(
+        indels_path,
+        force_bgz=True,
+        reference_genome="GRCh38",
+        contig_recoding=recode,
+        skip_invalid_loci=True,
+        min_partitions=1000,
+    )
+
+    spliceai_new_indels = hl.import_vcf(
+        new_indels_path,
+        header_file=header_file_path,
+        force_bgz=True,
+        reference_genome="GRCh38",
+        contig_recoding=recode,
+        skip_invalid_loci=True,
+        min_partitions=1000,
+    )
+
+    mt = spliceai_snvs.union_rows(spliceai_indels).union_rows(spliceai_new_indels)
+    ht = mt.rows()
+
+    logger.info("Exploding SpliceAI scores...")
+    # `explode` will eliminate rows with empty array, but the varaints with multiple genes will extend the number of rows.
+    ht = ht.explode(ht.info.SpliceAI)
+
+    logger.info("Annotating SpliceAI scores...")
+    # there will only be one gene in the array after exploding
+    gene_symbol = ht.info.SpliceAI.split(delim="\\|")[1:2][0]
+    delta_scores = ht.info.SpliceAI.split(delim="\\|")[2:6]
+    positions = ht.info.SpliceAI.split(delim="\\|")[6:10]
+    ht = ht.annotate(
+        splice_ai=hl.struct(
+            gene_symbol=gene_symbol,
+            delta_scores=hl.map(lambda x: hl.float32(x), delta_scores),
+            positions=hl.map(lambda x: hl.int32(x), positions),
+        )
+    )
+
+    # Annotate info.max_DS with the max of DS_AG, DS_AL, DS_DG, DS_DL in info.
+    # delta_score array is |DS_AG|DS_AL|DS_DG|DS_DL
+    logger.info(
+        "Getting the max SpliceAI score for each variant across consequences per"
+        " gene..."
+    )
+    consequences = hl.literal(
+        ["acceptor_gain", "acceptor_loss", "donor_gain", "donor_loss"]
+    )
+    ht = ht.annotate(
+        splice_ai=ht.splice_ai.annotate(ds_max=hl.max(ht.splice_ai.delta_scores))
+    )
+    ht = ht.annotate(
+        splice_ai=ht.splice_ai.annotate(
+            consequence_max=hl.if_else(
+                ht.splice_ai.ds_max > 0,
+                consequences[ht.splice_ai.delta_scores.index(ht.splice_ai.ds_max)],
+                "no_consequence",
+            ),
+            position_max=hl.if_else(
+                ht.splice_ai.ds_max > 0,
+                ht.splice_ai.positions[
+                    ht.splice_ai.delta_scores.index(ht.splice_ai.ds_max)
+                ],
+                hl.missing(hl.tint32),
+            ),
+        )
+    )
+
+    logger.info("Getting the max SpliceAI score for each variant across genes...")
```

Comment:
```suggestion
    logger.info("Getting the max SpliceAI score at each position across genes...")
```
I think this is more accurate since you are grouping by locus?

Reply:
grouping by both locus and alleles, it's not just by position.